cmd/image-builder/describeimg.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"slices"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/osbuild/images/pkg/blueprint"
+	"github.com/osbuild/images/pkg/distro"
+	"github.com/osbuild/images/pkg/imagefilter"
+)
+
+// Use yaml output by default because it is both nicely human and
+// machine readable and parts of our image defintions will be written
+// in yaml too.  This means this should be a possible input a
+// "flattended" image definiton.
+type describeImgYAML struct {
+	Distro string `yaml:"distro"`
+	Type   string `yaml:"type"`
+	Arch   string `yaml:"arch"`
+
+	// XXX: think about ordering (as this is what the user will see)
+	OsVersion string `yaml:"os_vesion"`
+
+	Bootmode        string `yaml:"bootmode"`
+	PartitionType   string `yaml:"partition_type"`
+	DefaultFilename string `yaml:"default_filename"`
+
+	// XXX: add pipelines here? maybe at least exports?
+	Packages *packagesYAML `yaml:"packages"`
+}
+
+type packagesYAML struct {
+	Include []string `yaml:"include"`
+	Exclude []string `yaml:"exclude"`
+}
+
+func packageSetsFor(imgType distro.ImageType) (inc, exc []string, err error) {
+	var bp blueprint.Blueprint
+	manifest, _, err := imgType.Manifest(&bp, distro.ImageOptions{}, nil, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	incM := map[string]bool{}
+	excM := map[string]bool{}
+	// XXX: or should this just do what osbuild-package-sets does
+	// and inlcude what pipeline needs the package set too?
```

Comment:
Personally, I think that merging pipeline package sets makes little sense because it will be confusing and impossible to determine what will be in the "payload" and what will be installed, e.g., in the Anaconda ISO for installer images.